analysis/src/type_check.rs

```diff
@@ -0,0 +1,74 @@
+use ast::{
+    analysis::{
+        AnalysisASMFile, AssignmentStatement, DegreeStatement, InstructionDefinitionStatement,
+        InstructionStatement, LabelStatement, PilBlock, RegisterDeclarationStatement,
+    },
+    parsed::asm::{ASMFile, ASMStatement},
+};
+use number::FieldElement;
+
+/// A very stupid type checker. TODO: make it smart
```

Comment:
Can you make an issue for it so we remember?

Reply:
#338